### ISSUE TITLE
Optimise add_pmfs

### ISSUE THREAD

**Issue by pearsonca** [epinowcast/epinowcast#186]
https://github.com/epinowcast/epinowcast/blob/1b39e77447d0b858afa810806b8984f0945f009b/R/model-module-helpers.R#L173

core replacment fits in a tweet:

```
Reduce(\(conv, pmf) {
    lc <- length(conv)
    wd <- (1:lc)-1
    proc <- numeric(lc + length(pmf))
    for (j in seq_along(pmf)) {
      proc[j+wd] <- proc[j+wd] + pmf[j] * conv
    }
    proc
  }, pmfs)
```

If this is actually hot in the performance chain (doubt it, but ... :shrug:), then can just replace that with some cpp.

@pratikunterwegs

**Comment by seabbs** [comment]
Nice. Nah it isn't as it's just prep for the stan modelling side of things (i.e gets called once prior to the much chunkier model fitting). 

Does this reproduce the tests/examples I am not entirely 💯 the indexes shake out. 

What version did the `\()` function syntax get added? I think it is pretty new so might want to avoid. 

Very happy to see this PR'd in 😄

**Comment by pratikunterwegs** [comment]
`\()` is from R 4.1.0 iirc, so indeed might want to avoid

**Comment by seabbs** [comment]
Closed by #187
